crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2377,6 +2377,255 @@ impl World {
+
+        let mut inserter_with_archetype: Option<(BundleInserter, ArchetypeId)> = None;
+
+        // `insert_or_spawn_batch_with_caller` does this implicitly with a `BundleSpawner`,
+        // but we only need the world cell here
+        let world_cell = self.as_unsafe_world_cell();
+
+        for (entity, bundle) in iter {
+            if let Some(location) = world_cell.entities().get(entity) {
+                match inserter_with_archetype {
+                    Some((ref mut inserter, archetype)) if location.archetype_id == archetype => {
+                        // SAFETY: `entity` is valid, `location` matches entity, bundle matches inserter
+                        unsafe {
+                            inserter.insert(
+                                entity,
+                                location,
+                                bundle,
+                                insert_mode,
+                                #[cfg(feature = "track_change_detection")]
+                                caller,
+                            )
+                        };
+                    }
+                    _ => {
+                        // SAFETY: we initialized this bundle_id in `register_info`
+                        let mut inserter = unsafe {
+                            BundleInserter::new_with_id(
+                                world_cell.world_mut(),
+                                location.archetype_id,
+                                bundle_id,
+                                change_tick,
+                            )
+                        };
+                        // SAFETY: `entity` is valid, `location` matches entity, bundle matches inserter
+                        unsafe {
+                            inserter.insert(
+                                entity,
+                                location,
+                                bundle,
+                                insert_mode,
+                                #[cfg(feature = "track_change_detection")]
+                                caller,
+                            )
+                        };
+                        inserter_with_archetype = Some((inserter, location.archetype_id));
+                    }
+                }
+            } else {
+                panic!("error[B0003]: Could not insert a bundle (of type `{}`) for entity {:?} because it doesn't exist in this World. See: https://bevyengine.org/learn/errors/b0003", core::any::type_name::<B>(), entity);
+            }
+        }
+    }
+
+    /// For a given batch of ([`Entity`], [`Bundle`]) pairs,
+    /// adds the `Bundle` of components to each `Entity`.
+    /// This is faster than doing equivalent operations one-by-one.
+    ///
+    /// This will overwrite any previous values of components shared by the `Bundle`.
+    /// See [`World::try_insert_batch_if_new`] to keep the old values instead.
+    ///
+    /// This function silently fails by ignoring any entities that do not exist.
```

Comment:
The `try_` prefix of this function makes me thing that it would return a `Result`, but instead it silently ignores errors.

How difficult is it to return an `Err` on failure, and is that desired at all?

Reply:
The existing `try_insert` functions all fail silently I believe. Maybe that could change, but they should change together

Reply:
Ok, I'm cool with keeping things consistent!